src/main/services/storage.js

```diff
@@ -0,0 +1,12 @@
+import Store from 'electron-store';
+import { ipcMain } from 'electron';
+
+const store = new Store();
+
+ipcMain.on('storage.delete', (event, key) => {
+  store.delete(key);
+});
+
+ipcMain.on('storage.set', (event, key, value) => {
+  store.set(key, value);
```

Comment:
Both set and delete here in this method end up writing back to the file system synchronously for every storage event:
See
https://github.com/sindresorhus/conf/blob/master/index.js#L70
https://github.com/sindresorhus/conf/blob/master/index.js#L156

We should really be batching writes and writing back if more than an expiry time expires or if we have more than 'x' number of writes.

Reply:
Writes happen in the main process via the IPC, so it won't affect the performance of the UI as they are on different processes/threads.

Reply:
Yes, this comment was not about affecting performance in the UI thread; my concern here is excessive writes to disk. For example, ten events in rapid succession writes the whole file 10 times. Do we know how often we are triggering these events and the impact of this once the storage size gets bigger. We are saving transaction history in storage, correct?